winml/test/adapter/AdapterDmlEpTest.cpp

```diff
@@ -158,10 +158,10 @@ void DmlGetD3D12ResourceFromAllocation() {
   auto session = CreateDmlSession();
   OrtExecutionProvider* ort_provider;
   THROW_IF_NOT_OK_MSG(winml_adapter_api->SessionGetExecutionProvider(session.get(), 0, &ort_provider), ort_api);
-  winrt::com_ptr<ID3D12Resource> d3d12_resource_from_allocation;
```

Comment:
Does `DmlGetD3D12ResourceFromAllocation` not properly increment the ref count of the output parameter? It's mandatory that COM functions increment the ref count of IUnknown derived output parameters. Are we sure that's not the real issue?

Reply:
Okay, I didn't know whether `DmlGetD3D12ResourceFromAllocation` was supposed to increment it and assumed it was a test bug. Do you want to merge the PR and revert the change once DML gets fixed? I'll add a comment that this is a DML bug

Reply:
But wait, `DmlGetD3D12ResourceFromAllocation` doesn't return a COM pointer, the last parameter is (`ID3D12Resource**`). I think it's a test bug.

Reply:
Dwayne showed me that `DmlGetD3D12ResourceFromAllocation` is actually in the WinML adapter :P I'll work on a fix.